arangod/Aql/AqlItemBlockSerializationFormat.h

```diff
@@ -0,0 +1,39 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2019-2019 ArangoDB GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Michael Hackstein
+////////////////////////////////////////////////////////////////////////////////
+
+#ifndef ARANGOD_AQL_AQLITEMBLOCK_SERIALIZATION_FORMAT_H
+#define ARANGOD_AQL_AQLITEMBLOCK_SERIALIZATION_FORMAT_H
+
+namespace arangodb {
+namespace aql {
+
+/// @brief Defines the serialization format of AQL item blocks
+enum class SerializationFormat {
+  // Format used in 3.5 and early. Not containing shadow rows
+  CLASSIC = 0,
+  // Use a hidden register for shadow rows. In classic versions all entries would be off by one.
+  SHADOWROWS = 1
```

Comment:
You have passed `SHADOWROWS` explicitly in many places. The only place where `CLASSIC` can be set is when the snippet gets deserialized. All places where `SHADOWROWS` is used will need to be updated in case a new version gets here, because they all just want the most current version, not explicitly `SHADOWROWS`.

I propose to add
```suggestion
  SHADOWROWS = 1,
  CURRENT = 1
```
and use that whenever you just want the "default". This can then be updated if a new version gets added. This should, of course, also be describe in detail in the comment above.

Reply:
agreed, will add this in separate PR.